app/main.py

```diff
@@ -1,25 +1,82 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self,
+                 min_amount: int,
+                 max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: str) -> Any:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: Any, value: Any) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: Any, value: Any) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Grade should be integer")
+        if self.min_amount > value or value > self.max_amount:
+            raise ValueError("Grade should not be less than "
+                             "2 and greater than 12")
```

Comment:
The error message in the `IntegerRange` class is hardcoded to 'Grade should not be less than 2 and greater than 12'. This message should dynamically reflect the `min_amount` and `max_amount` values to provide accurate feedback.